### PR TITLE
Fix up migration script and docs.

### PR DESCRIPTION
**Pull request by sfc-gh-dkurokawa**
# Description
Fix up migration script and docs.

## Other details good to know for developers
This should make it clearer on how to do DB migrations and also fix up some issues with it.

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] New Tests
- [ ] This change includes re-generated golden test results
- [ ] This change requires a documentation update

### CODE REVIEW

File: src/core/trulens/core/database/migrations/script.py.mako [hunk 1]
```diff
@@ -25,7 +25,7 @@ def upgrade(config) -> None:
     ${upgrades if upgrades else "pass"}
 
 
-def downgrade() -> None:
+def downgrade(config) -> None:
```
Comment (sfc-gh-chu) [inline src/core/trulens/core/database/migrations/script.py.mako:28]: Can we make the TODO clearer?

File: src/core/trulens/core/database/migrations/data.py [hunk 2]
```diff
@@ -16,41 +16,36 @@
 from trulens.core.schema import record as mod_record_schema
 from trulens.core.utils.pyschema import FunctionOrMethod
 
-sql_alchemy_migration_versions: List[str] = ["1"]
-"""DB versions that need data migration.
+sql_alchemy_migration_versions: List[int] = [1]
```
Comment (sfc-gh-chu) [inline src/core/trulens/core/database/migrations/data.py:19]: Does the typing play a role? Our migrations are ascending integers but are typed as strings and can be arbitrary strings

**Review by sfc-gh-chu — COMMENTED** [review]
Can we also add a migration script for the feedback function ID being not nullable? It's currently out of sync with the ORM

File: src/core/trulens/core/database/migrations/data.py (hunk 2 above)
Comment (sfc-gh-dkurokawa) [reply src/core/trulens/core/database/migrations/data.py:19]: Yes, but the code was actually assuming they were all ints anyway so I just made it more explicit.

**Comment by sfc-gh-dkurokawa** [comment]
> Can we also add a migration script for the feedback function ID being not nullable? It's currently out of sync with the ORM

Isn't this part of your app versioning PR?

**Review by sfc-gh-chu — APPROVED** [review]

**Comment by sfc-gh-dhuang** [comment]
> Can we also add a migration script for the feedback function ID being not nullable? It's currently out of sync with the ORM

One thing to note here is that I don't believe SQLite support ALTER a originally nullable column to be not nullable. For all I know SQLite ALTER statement is very limited and only supports basic stuff like adding a new column, dropping columns,  renaming a table / column.  

cc @sfc-gh-dkurokawa @sfc-gh-chu

**Pull request merged**